File: Medium/MaximizeHappinessofSelectedChildren.swift
class Solution {
    func maximumHappinessSum(_ happiness: [Int], _ k: Int) -> Int {
        let happyChildren = happiness.sorted(by: >)
        var happyChildrenSum = 0
        for i in 0..<k {
            happyChildrenSum += max(0, happyChildren[i] - i)
        }
        return happyChildrenSum
    }
}

File: Medium/MaximizeHappinessofSelectedChildren.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        happykids = 0
        for i in range(k):
            happykids += max(0, happiness[i] - i)
        return happykids
